$NetBSD: patch-aa,v 1.5 2004/07/15 13:27:48 mrg Exp $

- fix print when file sizes >= 2GB
- add "exit" and "^D" support

--- udfclient.c.orig	2004-02-23 11:10:49.000000000 +1100
+++ udfclient.c	2004-07-05 19:33:11.000000000 +1000
@@ -520,7 +520,7 @@ void udfclient_getsubtree(struct udf_mou
 			} else {
 				printf("\r...%-47s ", name+strlen(name)-47);
 			};
-			printf("%10d / %10d bytes ", (uint32_t) data_pos, (uint32_t) file_data_total_len);
+			printf("%10llu / %10llu bytes ", (unsigned long long) data_pos, (unsigned long long) file_data_total_len);
 			if (file_data_total_len) printf("(%3d%%) ", (int) (100.0*(float) data_pos / file_data_total_len));
 			printf("successfully");
 			fflush(stdout);
@@ -575,6 +575,9 @@ void udfclient_interact(void) {
 		*line = 0;
 		fgets(line, 4096, stdin);
 
+		if (*line == 0 && feof(stdin))
+			return;
+
 		*cmd = *arg1 = *arg2 = *arg3 = 0;
 		args = sscanf(line, "%s %s %s %s", cmd, arg1, arg2, arg3);
 		if (args) {
@@ -611,7 +614,8 @@ void udfclient_interact(void) {
 				      );
 				continue;
 			};
-			if (strcmp(cmd, "quit")==0) {
+			if (strcmp(cmd, "quit")==0 ||
+			    strcmp(cmd, "exit")==0) {
 				return;
 			};
 			printf("\nUnknown command %s\n", cmd);
